File: aarthik_setu/lib/models/profiles/personal_profile/components/contact_details.dart

class ContactDetails{
  String premisesName;
  String streetName;
  String landmark;
  String country;
  String state;
  String city;
  String pinCode;
  String village;
  String district;
  String subDistrict;
  String typeOfResidence;
  DateTime residenceSince;

  ContactDetails({
    required this.premisesName,
    required this.streetName,
    required this.landmark,
    required this.country,
    required this.state,
    required this.city,
    required this.pinCode,
    required this.village,
    required this.district,
    required this.subDistrict,
    required this.typeOfResidence,
    required this.residenceSince,
  });

  Map<String, dynamic> toJson() {
    return {
      'premisesName': premisesName,
      'streetName': streetName,
      'landmark': landmark,
      'country': country,
      'state': state,
      'city': city,
      'pinCode': pinCode,
      'village': village,
      'district': district,
      'subDistrict': subDistrict,
      'typeOfResidence': typeOfResidence,
      'residenceSince': residenceSince,
    };
  }
}

File: server/models/forms/personal_forms/contact_details.go
package personal_forms

import "time"

type ResidenceDetail struct {
	PremisesName      string    `form:"premisesName" json:"premises_name"`
	StreetName        string    `form:"streetName" json:"street_name"`
	Landmark          string    `form:"landmark" json:"landmark"`
	Country           string    `form:"country" json:"country"`
	State             string    `form:"state" json:"state"`
	City              string    `form:"city" json:"city"`
	PinCode           string    `form:"pinCode" json:"pin_code"`
	Village           string    `form:"village" json:"village"`
	District          string    `form:"district" json:"district"`
	SubDistrict       string    `form:"subDistrict" json:"sub_district"`
	TypeOfResidence   string    `form:"typeOfResidence" json:"type_of_residence"`
	ResidenceSince    time.Time `form:"residenceSince" json:"residence_since"`
}
